### PR TITLE
 interfaces/opengl: don't udev tag nvidia devices and use snap-confine instead (2.28)

### PR DESCRIPTION
**Pull request by mvo5**
This should fix the regression in the opengl interface with the nvidia driver.

This is the 2.28 version of https://github.com/snapcore/snapd/pull/3938

### CODE REVIEW

File: cmd/snap-confine/snap-confine.apparmor.in [hunk 1]
```diff
@@ -249,6 +249,14 @@
     # nvidia handling, glob needs /usr/** and the launcher must be
     # able to bind mount the nvidia dir
     /sys/module/nvidia/version r,
+    /sys/bus/pci/drivers/nvidia/uevent r,
+    /sys/module/nvidia_uvm/uevent r,
+    /sys/module/nvidia_modeset/uevent r,
+    /sys/module/nvidia_drm/uevent r,
+    /sys/module/nvidia/uevent r,
```
Comment (jdstrand) [inline cmd/snap-confine/snap-confine.apparmor.in]: I suggest for future-proofing:
```
/sys/module/nvidia/* r,
/sys/**/nvidia*/uevent r,
```

Comment (jdstrand) [reply cmd/snap-confine/snap-confine.apparmor.in]: Let's make this even better:
```
/sys/module/nvidia{,_*}/* r,
/sys/**/drivers/nvidia{,_*}/* r,
```

Comment (zyga) [reply cmd/snap-confine/snap-confine.apparmor.in]: I applied both in sequence.

File: cmd/snap-confine/snap-confine.apparmor.in [hunk 2]
```diff
@@ -249,6 +249,12 @@
     # nvidia handling, glob needs /usr/** and the launcher must be
     # able to bind mount the nvidia dir
     /sys/module/nvidia/version r,
+    /sys/**/drivers/nvidia{,_*}/* r,
+    /sys/**/nvidia*/uevent r,
+    /sys/module/nvidia{,_*}/* r,
```
Comment (jdstrand) [inline cmd/snap-confine/snap-confine.apparmor.in]: This covers the first rule (/sys/module/nvidia/version r,) so you can omit the first rule.

Comment (zyga) [reply cmd/snap-confine/snap-confine.apparmor.in]: Let's remove this in the post-mortem move and let mvo release the point release.

**Pull request merged**